static/js/unread.js

```diff
@@ -29,6 +33,28 @@ export function set_messages_read_in_narrow(value) {
 
 export const unread_mentions_counter = new Set();
 const unread_messages = new Set();
+// Key is "stream:topic" and values (type Set) are the messages with unread mentions.
+export const unread_mention_topics = new Map();
+
+function add_message_to_unread_mention_topics(message_id) {
+    const message = message_store.get(message_id);
+    const topic_key = recent_topics_util.get_topic_key(message.stream_id, message.topic);
+    if (unread_mention_topics.has(topic_key)) {
+        unread_mention_topics.get(topic_key).add(message_id);
+    }
+    unread_mention_topics.set(topic_key, new Set([message_id]));
+}
+
+function remove_message_from_unread_mention_topics(message_id) {
+    const message = message_store.get(message_id);
+    if (message.type !== "stream") {
+        return;
+    }
```

Comment:
Probably this detail should be symmetric with the `add_message_...` code?

Reply:
Ok, added it there too.